opentelemetry-sdk/src/metrics/meter_provider.rs

```diff
@@ -131,15 +134,13 @@ impl SdkMeterProviderInner {
         }
     }
 
-    fn shutdown(&self) -> MetricResult<()> {
+    fn shutdown(&self) -> ShutdownResult {
         if self
             .shutdown_invoked
             .swap(true, std::sync::atomic::Ordering::SeqCst)
         {
             // If the previous value was true, shutdown was already invoked.
-            Err(MetricError::Other(
-                "MeterProvider shutdown already invoked.".into(),
-            ))
+            Err(crate::error::ShutdownError::AlreadyShutdown)
```

Comment:
I'm still not convinced that this should be an error, rather than just asserting that `shutdown` should be idempotent. @cijothomas I feel like you might've had a good argument for this I've forgotten?

Reply:
https://github.com/open-telemetry/opentelemetry-rust/issues/2571#issuecomment-2625145109 As discussed, we agree to keep this as-is. This is done so that user can realize that someone else in their codebase is calling shutdown before them, and that is a mistake/issue they need to rectify. Given this is user actionable we should keep this.

(Also, internal logging may not help always when we are at the shutdown phase, as ordering of shutdowns may cause internal logs to stop working when we get to shutdown!!)